File: Data Structures & Algorithms/validate-parentheses/submission-2.cs
public class Solution {
    public bool IsValid(string s) {
        var mp = new Dictionary<char, char>();
        mp[')'] = '(';
        mp['}'] = '{';
        mp[']'] = '[';
        var stack = new Stack<char>();
        for(int i=0;i<s.Count();i++){
            if(!mp.ContainsKey(s[i])) stack.Push(s[i]);
            else if(stack.Count()>0 && mp[s[i]]==stack.Peek()) stack.Pop();
            else return false;
        }
        return stack.Count()==0;
    }
}

File: Data Structures & Algorithms/evaluate-reverse-polish-notation/submission-2.java
class Solution {
    public int evalRPN(String[] tokens) {
        Stack<Integer> stack = new Stack();
        for(int i=0;i<tokens.length;i++){
            int a = 0, b = 0;
            switch(tokens[i]){
                case "+":
                    a = stack.pop();
                    b = stack.pop();
                    stack.push(b+a);
                    break;
                case "-":
                    a = stack.pop();
                    b = stack.pop();
                    stack.push(b-a);
                    break;
                case "*":
                    a = stack.pop();
                    b = stack.pop();
                    stack.push(b*a);
                    break;
                case "/":
                    a = stack.pop();
                    b = stack.pop();
                    stack.push(b/a);
                    break;
                default:
                    stack.push(Integer.parseInt(tokens[i]));
                    break;
            }                   
        }
        return stack.peek();
    }
}
